build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java

```diff
@@ -107,16 +114,10 @@ public static Properties stripPrefixFromProperties(Properties source, String pre
     }
 
     public static Properties getEnvProperties(Properties startProps, Log log) {
-        IncludeExcludePatterns patterns = new IncludeExcludePatterns(
```

Comment:
Removing the filtering code from this method sounds very risky to me. Let's do the "get" and the "filter" logic in one place to enforce filtering in all flows.

Reply:
I understand the concern, but this is not the place to do filtering. According to the name of this function, it returns all environment variables. By moving the filter logic to a new location, we can filter not only the environment variables, but also modules. Putting the filter logic here would prevent us from filtering things unrelated to environment variables, so we would have to split the logic across the code, and that will cause a mess in the code.

Reply:
Due to security risks, we will return the filter logic to here despite the above comment .